src/main/java/frc/robot/hardware/request/cansparkmax/SparkMaxRequestBuilder.java

```diff
@@ -0,0 +1,29 @@
+package frc.robot.hardware.request.cansparkmax;
+
+import com.revrobotics.CANSparkBase;
+import edu.wpi.first.math.geometry.Rotation2d;
+
+import java.util.function.Function;
+
+public class SparkMaxRequestBuilder {
+
+	//@formatter:off
+    public SparkMaxRequest<Rotation2d> generate(
```

Comment:
If it's a "builder" wouldn't these be called `build()`? With the current name this seems like a `Generator`. Actually I don't see why not just add these as static functions of `SparkMaxRequest`, I think that's cleaner

Reply:
To clarify, you'd write `SparkMaxRequest.with(args...)` (or whatever name you choose).
If you want to be extra explicit with named constructors, you can have `.revolutionsPerSecond(Rotation2d)`, `.voltage(Double)`, and `.amperage(Double)` (keeping the control type as more of an implementation detail)

Reply:
It cannot have multiple setpoints, so how would it work

Reply:
those would be static methods, not instance methods. But indeed the name `with` might be confusing then. Probably `.of` is fine or `construct` or `newVoltage`/`newRPS`/...

Reply:
u r breaking SRP mi friendo. It does not really matter but i prefer it as diffrent builder

Reply:
yeah that's fine. Just make sure the method and class names match up